plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorService.java

```diff
@@ -53,8 +56,34 @@ public void run() {
      * That makes it worth reporting.
      */
     private void reportProblem(Throwable t) {
+        if (isShutdown()) {
+            // We probably already got here once with the actual root cause.
+            LOGGER.log(Level.FINE, t, () -> "Unexpected exception in CPS VM thread: " + cpsThreadGroup.getExecution());
+            return;
+        }
         LOGGER.log(Level.WARNING, "Unexpected exception in CPS VM thread: " + cpsThreadGroup.getExecution(), t);
+        try {
+            // Give steps a chance to clean up.
+            for (CpsThread thread : cpsThreadGroup.getThreads()) {
+                StepExecution se = thread.getStep();
+                if (se != null) {
+                    try {
+                        se.stop(t);
+                        TaskListener listener = se.getContext().get(TaskListener.class);
+                        FlowNode node = se.getContext().get(FlowNode.class);
+                        if (listener != null && node != null) {
+                            listener.getLogger().println("Terminating " + node.getDisplayFunctionName() + " (id: " + node.getId() + ")");
+                        }
+                    } catch (Exception e) {
+                        t.addSuppressed(e);
+                    }
+                }
+            }
+        } catch (Throwable t2) {
+            t.addSuppressed(t2);
+        }
```

Comment:
I think if we switch the inner catch to `Throwable` this should be unreachable and could be deleted.

Reply:
Simplified in [e76ebb6](https://github.com/jenkinsci/workflow-cps-plugin/pull/780/commits/e76ebb65eb3178d6a21599ce23d72ac616dec9d8).